### DISCUSSION TITLE
Battery entities for energy dashboard

### DISCUSSION THREAD

**Discussion by adam-cobb** [macxq/foxess-ha#280] [Q&A] [answered]
Apologies if someone has asked this before but i've just had solar PV and Fox inverter/battery installed and setting up this integration (thanks btw!).

I have read the wiki articles about better sensors for solar PV generation and plan to add those, but i'm not sure what I should be using in the HA energy dashboard for the battery. The dashboard wants energy going in/out as KWh sensors, but as far as I can see the ones the integration provides battery charge power and battery discharge power as instant KW values.

I have a feeling this is a thing, in terms of converting between the two but whenever I read up about this I have to admit I struggle to understand it fully!

Any help appreciated, thanks :)

**Comment by Daibutt** [comment]
Not had this integratiion installed for quite a while but don't you have the following to entities/sensors;
![image](https://github.com/user-attachments/assets/9ef35f1f-fa3d-4f21-a34e-72d4fd83bb34)

**Reply by adam-cobb** [reply]
Hmm no I don't see any of those,  just sensors called Battery Charge Power and Battery Discharge Power, which are in KW. Unless I'm missing something!

Doesn't appear to be any additional disabled entities from what I can see

**Reply by adam-cobb** [reply]
Ah but I do have sensors called Daily Charged Energy Total and Daily Discharged Energy Total. I wonder if I can use those

**Comment by Daibutt** [comment]
Yeah def worth a try. I'm not aware of those sensors, but if the dashboard
accepts them see what happens. It may take a few hours till you see
anything in the dashboard remember.

**Comment by FozzieUK** [comment, marked as answer]
@adam-cobb  Yes Dave is spot on, those sensors come directly from the Fox reports and provide the daily totals for battery charge and discharge, likewise Grid Consumption and Feedin sensors which can all be added directly to the energy dashboard as required.
